docs/DeveloperGuide.md

```diff
@@ -346,74 +346,177 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
-`* *` | user | hide [private contact details](#private-contact-detail) by default | minimize chance of someone else seeing them by accident
-`*` | user with many persons in the address book | sort persons by name | locate a person easily
+`* * *` | user | add tasks | keep track of them
+`* * *` | user | be able to edit a task after adding it | make any changes if I need to
+`* * *` | user | be able to delete a task after adding | delete tasks that I no longer need
+`* * *` | user | find specific tasks by keywords | locate my task easily
+`* * *` | user | specify a particular path or folder to store my task lists | retrieve the file easily and conveniently
+`* * *` | user | undo the recent action | revert back in case I decide to change my mind.
+`* *` | user | add tasks that recur on a consistent basis (e.g. Tutorials and lectures) | add recurring tasks more conveniently, since the item only needs to be added once
+`* *` | user | add some comments for a particular task | store additional information that I will need when handling that task
+`* *` | user | include a tag on the task | retrieve all the tasks that belong to the same tag
+`* *` | user | be able to set how important a task is | know which tasks I should prioritize first
+`* *` | user | sort my tasks according to priority or tags | delegate my time appropriately
+`* *` | user | be able to delete a group of task by keywords | avoid deleting one by one
+`* *` | user | have a view of my pending tasks for the day/week/month | know what I need to do for that particular period of time
+`* *` | user | see my overdue tasks | know what actions to take on them
+`* *` | advanced user | have the choice of a few preset themes | change it to my liking.
+`*` | user | sort my tasks according to task names | delegate my time appropriately
+`*` | user | delete all my tasks | hand in my resignation letter afterwards
+
+
+## Appendix B: Use Cases
+#### Use case: Create a task
 
-{More to be added}
+**MSS**
+
+1. User inputs the command with the correct keywords
+2. Task Manager confirms the addition by showing the confirmation message <br>
+
+> Use case ends.
 
-## Appendix B : Use Cases
+**EXTENSIONS**
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+2a. The given input is invalid
 
-#### Use case: Delete person
+> 2a1. Task Manager shows an error message <br>
+Use case ends
+
+#### Use case: Find specific tasks
 
 **MSS**
 
-1. User requests to list persons
-2. AddressBook shows a list of persons
-3. User requests to delete a specific person in the list
-4. AddressBook deletes the person <br>
-Use case ends.
+1. User inputs the command with the correct keywords
+2. Task Manager shows the list of tasks that match the input <br>
+
+> Use case ends.
+
+**EXTENSIONS**
 
-**Extensions**
+2a. The given input is invalid
 
-2a. The list is empty
+> 2a1. Task Manager shows an error message <br>
+Use case ends
+
+3a. The list is empty
```

Comment:
would be nice if you can display "No tasks found" instead of just simply leaving it there